tango/integrations/beaker/workspace.py

```diff
@@ -291,8 +289,8 @@ def _get_run_from_dataset(self, dataset: Dataset) -> Optional[Run]:
         try:
             run_name = dataset.name[len(Constants.RUN_DATASET_PREFIX) :]
             steps: Dict[str, StepInfo] = {}
-            steps_info_bytes = b"".join(
-                list(self.beaker.dataset.stream_file(dataset, Constants.RUN_DATA_FNAME, quiet=True))
+            steps_info_bytes = self.beaker.dataset.get_file(
```

Comment:
`get_file()` is a little more robust than `stream_file()` because it will treat `ChecksumFailed` as a recoverable error and retry the whole download. That's not possible with `stream_file()`.